zetaclient/chains/evm/observer/outbound.go

```diff
@@ -366,9 +365,22 @@ func (ob *Observer) checkConfirmedTx(txHash string, nonce uint64) (*ethtypes.Rec
 		return nil, nil, false
 	}
 	if from != ob.TSS().EVMAddress() { // must be TSS address
-		log.Error().Msgf("confirmTxByHash: sender %s for outbound %s chain %d is not TSS address %s",
+		// If from is not TSS address, check if it is one of the previous TSS addresses We can still try to confirm a tx which was broadcast by an old TSS
+		log.Info().Msgf("confirmTxByHash: sender %s for outbound %s chain %d is not current TSS address %s",
 			from.Hex(), transaction.Hash().Hex(), ob.Chain().ChainId, ob.TSS().EVMAddress().Hex())
-		return nil, nil, false
+		addressList := ob.TSS().EVMAddressList()
+		isOldTssAddress := false
+		for _, addr := range addressList {
+			if from == addr {
+				isOldTssAddress = true
+			}
+		}
+		if !isOldTssAddress {
```

Comment:
I'm thinking of a concrete rule to verify the correctness of outbound `sender` address

- It's supposed to be one of the TSS addresses.
- We already have each TSS address and the nonce range that they ever worked on [here](https://zetachain-testnet-api.itrocket.net/zeta-chain/observer/pendingNonces). Not sure why BTC chain 18332 is still 0.
- We already have the `nonce` of the transaction.
- So the transaction's `sender` being checked here must equal to the TSS address that is responsible for that nonce range.

It's a future improvement if you can create a issue to track. `sender address be on the TSS address list` is a wider range than expected.

Reply:
I think thats a good idea , ill create a issue to track it 
'

Reply:
https://github.com/zeta-chain/node/issues/2487